libqtile/command/base.py

```diff
@@ -55,11 +57,118 @@ class CommandException(Exception):
 class CommandObject(metaclass=abc.ABCMeta):
     """Base class for objects that expose commands
 
-    Each command should be a method named `cmd_X`, where X is the command name.
+    Any command to be exposed should be decorated with
+    `@CommandObject.expose()` (classes that are not explicitly
+    inheriting from CommandObject will need to import the module)
     A CommandObject should also implement `._items()` and `._select()` methods
     (c.f. docstring for `.items()` and `.select()`).
     """
 
+    def __new__(cls, *args, **kwargs):
+
+        # Check which level of command object has been parsed
+        # This test ensures inherited classes don't stop additional
+        # methods from being exposed.
+        # For example, if widget.TextBox has already been parsed, a subsequent
+        # call to initialise a new TextBox will return here. However, if a user
+        # subclasses TextBox for a new widget then that new widget will still
+        # be parsed here to check for new commands.
+        if getattr(cls, "_command_object", "") == cls.__name__:
+            return super(CommandObject, cls).__new__(cls)
+
+        commands = {}
+        cmd_s = set()
+
+        # We need to iterate over the class's inherited classes in reverse order
+        # We reverse the order so the exposed command will always be the latest
+        # definition of the method.
+        for c in reversed(list(cls.__mro__)):
+            base_classes = []
+
+            for method_name in list(c.__dict__.keys()):
+                method = getattr(c, method_name, None)
+
+                if method is None:
+                    continue
+
+                # Ignore methods that were defined by subclasses
+                for b in base_classes:
+                    if method == getattr(b, method_name, None):
+                        continue
+
+                # If the command has been exposed, add it to our dictionary
+                # If the method name is already in our dictionary then bind the
+                # latest definition to that command
```

Comment:
Sorry if I'm missing something but I don't think that's a bug. I think the reverse MRO parsing is appropriate. Let's say a widget defines an `info` method but forgets to add the decorator. As `info` is decorated in the base class, the command is still exposed but, because of the reverse MRO parsing, the method that's bound to the command is the child widget's definition.